File: submission/12017800/code/01_basic_de/alpineFunction.cpp
#include "alpineFunction.h"

#include <cfloat>
#include <cmath>

using namespace std;

AlpineFunction::AlpineFunction() {

}

AlpineFunction::~AlpineFunction() {

}

long double AlpineFunction::getFitness(Individual* individual) {
	long double sinPart = 1;
	long double prodPart = 1;

	for (int i = 0; i < individual->getNumCells(); i++) {
		sinPart *= sin((*individual)[i].getValue());
		prodPart *= (*individual)[i].getValue();
	}

	if (prodPart < 0) {
		return FLT_MAX;
	}
	else {
		return sinPart * sqrt(prodPart);
	}
}

float AlpineFunction::getMin() {
	return -10;
}

float AlpineFunction::getMax() {
	return 10;
}

string AlpineFunction::getName() const {
	return "Alpine";
}

ostream& operator<<(ostream& os, const AlpineFunction& function) {
	os << function.getName();
	return os;
}

File: submission/12017800/code/04_random_coop/AlpineFunction.java
public class AlpineFunction implements OptimizationFunction {
	public double getFitness(Individual individual) {
		double sinPart = 1;
		double prodPart = 1;

		for (int i = 0; i < individual.getNumCells(); i++) {
			sinPart *= Math.sin(individual.getCell(i).getValue());
			prodPart *= individual.getCell(i).getValue();
		}

		if (prodPart < 0) {
			return Double.MAX_VALUE;
		}
		else {
			return sinPart * Math.sqrt(prodPart);
		}
	}

	public double getMin() {
		return -10.0;
	}
	
	public double getMax() {
		return 10.0;
	}

	@Override
	public String toString() {
		return "Alpine [" + getMin() + "," + getMax() + "]";
	}
}